src/components/grid/grid-container/grid-container.component.js

```diff
@@ -1,31 +1,13 @@
 import React from "react";
 import PropTypes from "prop-types";
+import propTypes from "@styled-system/prop-types";
 import GridContainerStyle from "./grid-container.style";
 import GridItem from "../grid-item";
 
 const GridContainer = (props) => {
-  const { children, p, pl, pr, pt, pb, px, py, gridGap } = props;
```

Comment:
So here you should leave it as `const { children, ...rest } = props` and pass `{...rest}` into `GridContainerStyle` rather than `{...props}`